[docs] Fargate: explain Wave (#5562)
nextflow-io/nextflow: docs/aws.md

and the Batch Execution Role.

:::{note}
Nextflow uses [s5cmd](https://github.com/peak/s5cmd) to download the task input data and upload the task outputs.

> To enable this capability, you need to enable the Wave service in the Nextflow configuration, as shown in the above example. See {ref}Wave <wave-page> documentation for more details.

Reviewer: Link is broken:
```
See {ref}`Wave <wave-page>` documentation for more details.
```

Reviewer 2: tagging @christopher-hakkaart 🙏

Author: Good spot, thanks @bentsherman .

PR to fix typo here: https://github.com/nextflow-io/nextflow/pull/5572

> Nextflow uses the [s5cmd](https://github.com/peak/s5cmd) to download the task input data and upload the task outputs.
> To allow this capability in your pipeline, you need to enable the Wave service in the Nextflow configuration, as shown in the above example. See {ref}Wave <wave-page> documentation for more details.

Reviewer 3: ```suggestion
Nextflow uses [s5cmd](https://github.com/peak/s5cmd) to download the task input data and upload the task outputs.
To enable this capability in your pipeline, you need to enable the Wave service in the Nextflow configuration, as shown in the above example. See {ref}Wave <wave-page> documentation for more details.
```

:::{note}

> This feature requires the use {ref}`Wave <wave-page>` container provisioning service.
> This is because `s5cmd` is used to stage and unstage data which is augmented into the task container via wave.

Reviewer 3: ```suggestion
This feature requires the {ref}`Wave <wave-page>` container provisioning service.
This is because `s5cmd` is used to stage and unstage data that is augmented into the task container via wave.
```

Reviewer 2: I'd improve like this 

> :::{note}
Under the hood Nextflow uses the [s5cmd](https://github.com/peak/s5cmd) to download the task input data
and upload the task outputs. To allow this capability your pipeline need to enable the Wave service 
in the Nextflow configuration as shown in the above example. See {ref}`Wave <wave-page>` documentation for more details.
:::

> :::{note}
> This feature requires the use {ref}`Wave <wave-page>` container provisioning service.
> This is because `s5cmd` is used to stage and unstage data which is augmented into the task container via wave.
> :::

Author: Suggested change by @pditommaso (some minor tweaks by me):
```suggestion
:::{note}
Nextflow uses the [s5cmd](https://github.com/peak/s5cmd) to download the task input data and upload the task outputs.
To allow this capability in your pipeline, you need to enable the Wave service in the Nextflow configuration, as shown in the above example. See {ref}Wave <wave-page> documentation for more details.
:::
```

That ok @christopher-hakkaart ?

Author: Committing so you can comment better
